### DISCUSSION TITLE
save station as yaml? ( or load station from snapshot? )

### DISCUSSION THREAD

**Discussion by silky** [microsoft/Qcodes#3335] [General]
I note that it's possible to _load_ a station from a yaml; but is it possible to _save_ a station to a yaml? I can't find a function in the API to do that? or is there some simple scheme to convert from the json snapshot to the yaml ( also, at a glance, it seems no; it seems like it'd be a small amount of work? i.e. the keys aren't the same?

**Comment by FarBo** [comment]
As far as I know, there is no built-in function to save a station to a yaml file and in my opinion, unless you have some specific use-case, there is no need for that. You can always add created instrument to the loaded station file via add_component function, but it only adds to the qcodes station object and not in the loaded station.yaml. Usually, users create a station yaml file before performing any experiment and put their entire experimental setup there. They only load specific instruments and parameters from their station file in their specific experiment.
Creating station.yaml is fairly easy using auto-completion in VS Code  if you need to fill it with a lot of instruments in the file:
[Setting up Visual Studio Code to work with station files](https://qcodes.github.io/Qcodes/examples/Station.html#Setting-up-Visual-Studio-Code-to-work-with-station-files)
I know this may not directly answer your issue, but I hope it would be somehow helpful.
